pop2vec/utils/parquet_walks.py

```diff
@@ -0,0 +1,140 @@
+from __future__ import annotations
+import pickle
+from collections import OrderedDict
+from dataclasses import dataclass
+from pathlib import Path
+from typing import TYPE_CHECKING
+import duckdb
+import numpy as np
+from pop2vec.utils.save_to_parquet import save_to_parquet
+
+if TYPE_CHECKING:
+    import pandas as pd
+
+
+def create_column_placeholders(cols) -> str:  # suggested by Claude
+    """Return comma-separated columns to query."""
+    return ", ".join(cols)
+
+
+@dataclass
+class ParquetWalks:
+    """Container for loading dataframes of walks for deepwalk."""
+
+    parquet_root: str
+    parquet_nests: str  # TODO: could this not be inferred from all the partitions that are arguments below?
+    year: int
+    iter_name: str
+    record_edge_type: bool
+    chunk_id: int | None = None
+    n_edge_types: int = 5  # hard-coded number of edge types. should find way to determine automatically
+
+    def remap_ids(self, mapped_indices):  # TODO: don't hard-code path here
+        """Remap an array of mapped indices.
+
+        From a list of mapped indices from 0 to len(mapped_indices)-1,
+        return the ids corresponding to the indices.
+        """
+        mapping_url = "/gpfs/ostor/ossc9424/homedir/data/graph/mappings/family_" + str(self.year) + ".pkl"
```

Comment:
this needs to go and be read in dynamically by year and data_config

Reply:
fixed in https://github.com/odissei-lifecourse/life-sequencing-dutch/pull/110/commits/2a5c55bec46db9afe02b1109b8ed8170deb961b8